elm/src/Types.elm

```diff
@@ -0,0 +1,98 @@
+module Types exposing (..)
+
+import RemoteData exposing (WebData)
+import Json.Decode
+import Json.Decode.Pipeline exposing (required)
+
+
+---- MODEL ----
+
+
+type Route
+    = HomePage
+    | NotFoundPage
+    | PostPage CrudRoute
+    | CommentPage CrudRoute
+
+
+type CrudRoute
+    = CrudNew
+    | CrudEdit Int
+    | CrudShow Int
+    | CrudList
+
+
+type alias Flags =
+    {}
+
+
+type alias Model =
+    { route : Route
+    , post : PostModel
+    , comment : CommentModel
+    }
+
+
+type alias PostModel =
+    { searchPosts : WebData (List Post)
+    , postByID : WebData Post
+    , formPost : Post
```

Comment:
Can consider having these `WebData` as part of `Crud*` routes, then each `{Table}Model` will not need to manage these states

Doing `CrudList WebData (List Post)` is fine, however, trying to do the same for `CrudNew`, `CrudEdit`, and `CrudShow` gets hairy